ttnn/cpp/ttnn/operations/data_movement/concat/concat.cpp

```diff
@@ -16,97 +16,101 @@ namespace ttnn {
 namespace operations {
 namespace data_movement {
 
-    // Wrapper for TTDNN
-    ttnn::Tensor ConcatOperation::invoke(
-        uint8_t queue_id,
-        const std::vector<ttnn::Tensor>& input_tensors,
-        int dim,
-        const std::optional<MemoryConfig>& memory_config,
-        std::optional<ttnn::Tensor> optional_output_tensor) {
-        TT_FATAL(input_tensors.size() > 0, "ttnn.concat: expected a non-empty list of Tensors!");
-        TT_FATAL(!optional_output_tensor.has_value(), "optional output tensor currently unsupported!");
-        const auto mem_config = memory_config.value_or(ttnn::DRAM_MEMORY_CONFIG); // should match input tensor memory config when unpopulated but causes CI errors for now
-
-        if (input_tensors.size() == 1) {
-            return ttnn::to_memory_config(input_tensors.at(0), mem_config, std::nullopt);
-        }
+// Wrapper for TTDNN
+ttnn::Tensor ConcatOperation::invoke(
+    uint8_t queue_id,
+    const std::vector<ttnn::Tensor>& input_tensors,
+    int dim,
+    const std::optional<MemoryConfig>& memory_config,
+    std::optional<ttnn::Tensor> optional_output_tensor,
+    unsigned int groups) {
```

Comment:
usually we prefer other parameters before optional_output_tensor and memory_config

Reply:
I put it last to avoid breaking the current interface. For example, in argmax:

```cpp
Tensor concat_out = ttnn::concat(combined_tensors, dim, output_memory_config);
```

Would have to become:

```cpp
Tensor concat_out = ttnn::concat(combined_tensors, dim, 1, output_memory_config);
```

I figured since groups is probably the least used param I put it at the end.

Reply:
Let me know what's preferred.